pkg/daemon/constants/constants.go

```diff
@@ -14,6 +14,8 @@ const (
 	DesiredMachineConfigAnnotationKey = "machineconfiguration.openshift.io/desiredConfig"
 	// MachineConfigDaemonStateAnnotationKey is used to fetch the state of the daemon on the machine.
 	MachineConfigDaemonStateAnnotationKey = "machineconfiguration.openshift.io/state"
+	// NodeOnHoldAnnotationKey is used by external components to put a node on hold from being accounted for in the node controller
```

Comment:
can we also mention something about "hold a node from being targeted for an upgrade/config change" from your pr desc here too?